### ISSUE TITLE
Name is Required

### ISSUE THREAD

**Issue by thewalrusisben** [lennym/busboy-body-parser#23]
Hey there, I ran across an issue upgrading to version 0.3.1 that was pretty challenging to understand, and should probably be noted somewhere in the package documentation. It turns out that `name` is now a required property of any file processed by busboy per [this update](https://github.com/lennym/busboy-body-parser/blob/master/index.js#L45).

For me, this came about because my client application is using node's [readfile](https://nodejs.org/api/fs.html#fs_fs_readfile_path_options_callback) to grab a local file which it then attempts to upload. By default, it seems, readfile doesn't automatically provide a `name` property to the file, so you have to assign one yourself. This in turn was resulting in the file coming out as an empty object after being rejected by busboy on the API.

Hope this helps someone else - took me forever to figure out!

**Comment by lennym** [comment]
Thanks for raising an issue on this - and potential thanks on behalf of future googlers 😄 

Would you be able to provide a minimal reproduction case? Purely so I can get a better understanding of what the use-case looks like and can provide either better documentation, error messaging, or mitigation as is appropriate. 

Edit: All I really need is a "how to construct a request that has a file with no name" case - it can be as little a curl command if that's all you need.

**Comment by thewalrusisben** [comment]
No problem, here you go!

server:
```
const app = require('express')()
const busboyBodyParser = require('busboy-body-parser')
app.use(busboyBodyParser())

app.post('/foo', (req) => {
  console.log(req.files)
})

app.listen(3000, console.log('👋'))
```

client:
```
const fs = require('fs')
const request = require('request')

fs.readFile('./img.jpg', (error, file) => {
  file.name = 'foo' // If you comment this out, you'll see in your server that req.files is an empty object!
  request.post({url: `http://127.0.0.1:3000/foo`,formData: {file}})
})
```

Hope that helps, if not let me know!

**Comment by James1x0** [comment]
This might actually be an issue, as assigning a file name is just a workaround. Maybe we need a different way of checking for an empty payload, instead of assuming name signifies an empty pass...

See: https://github.com/lennym/busboy-body-parser/blob/master/index.js#L45

**Comment by lennym** [comment]
Thanks. I'll try to find some time to take a look at this issue asap.

Happy to review PRs in the meantime if anyone has time to resolve.

**Comment by lennym** [comment]
Thanks for your comments. I've had a look again, and it is literally just that line. IIRC that was there to catch the case when an html form with a 'file' input was submitted with no file selected. The server would receive an _empty_ file, so this was to prevent that being registered as an actual file.

I'll come up with a better way to do that though. Expect a release later today (UK time).

**Comment by lennym** [comment]
That was actually pretty easy. Published a fix at v0.3.2.

@thewalrusisben Thanks for raising the issue, and please accept my apologies for taking so long to fix.

**Comment by thewalrusisben** [comment]
no problem, and please accept my delayed expression of gratitude for fixing @lennym  :)
